tools/generator/src/Generator+ConsolidateTargets.swift

```diff
@@ -0,0 +1,208 @@
+import OrderedCollections
+import XcodeProj
+
+/// Collects multiple Bazel targets (see `Target`) that can be represented by
+/// a single Xcode target (see `PBXNativeTarget`).
+///
+/// In a Bazel build graph there might be multiple configurations of the
+/// same label (e.g. macOS and iOS flavors of the same `swift_library`).
+/// Xcode can represent these various configurations as a single target,
+/// using build settings, and conditionals on build settings, to account
+/// for the differences.
+struct ConsolidatedTarget: Equatable {
+    let name: String
+    let label: String
+    let product: ConsolidatedTargetProduct
+    let isSwift: Bool
+    let resourceBundles: Set<FilePath>
+    let inputs: ConsolidatedTargetInputs
+    let linkerInputs: ConsolidatedTargetLinkerInputs
+    let outputs: ConsolidatedTargetOutputs
+
+    /// The `Set` of `FilePath`s that each target references above the baseline.
+    ///
+    /// The baseline is all `FilePath`s that each target references. A reference
+    /// in this case is anything that the `EXCLUDED_SOURCE_FILE_NAMES` and
+    /// `INCLUDED_SOURCE_FILE_NAMES` apply to.
+    let uniqueFiles: [TargetID: Set<FilePath>]
+
+    let targets: [TargetID: Target]
```

Comment:
Is it ever important to keep the original unsorted target list? It looks like we mostly just loop over the targets. Perhaps as a simplification, we could consolidate by having `targets` be the sorted list and removing `sortedTargets`.

Reply:
Yes. We need the dictionary functionally in some cases later.